File: leetcode/javascript/maximum-length-of-pair-chain.js
var findLongestChain = function (pairs) {
    pairs.sort((a, b) => a[1] - b[1]);
    let current = Number.MIN_SAFE_INTEGER, answer = 0
    for(const [start,end] of pairs){
        if(start > current){
            answer+=1;
            current = end
        }
    }
    return answer
}

File: leetcode/cpp/maximum-length-of-pair-chain.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int findLongestChain(vector<vector<int>> pairs)
    {
        sort(pairs.begin(), pairs.end(), [](const vector<int> &a, const vector<int> &b)
             { return a[1] < b[1]; });
        int current = INT_MIN, answer = 0;
        for(vector<int> pair: pairs){
            if(pair[0] > current){
                answer++;
                current = pair[1];
            }
        }
        return answer;
    }
};